### PR TITLE
Add new `Style/SingleLineDoEndBlock` cop

### PR DESCRIPTION
**Pull request by koic**
This PR adds new `Style/SingleLineDoEndBlock` cop that checks for single-line `do`...`end` blocks.

```ruby
# bad
foo do |arg| bar(arg) end

# good
foo do |arg|
  bar(arg)
end

# bad
->(arg) do bar(arg) end

# good
->(arg) { bar(arg) }
```

In practice a single line `do`...`end` is autocorrected when `EnforcedStyle: semantic` in `Style/BlockDelimiters`. It can also be detected by this cop if it is written by handcraft. So I decided to introduce this new cop instead of an autocorrect extension of `Style/BlockDelimiters`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -5186,6 +5186,11 @@ Style/SingleLineBlockParams:
         - acc
         - elem
 
+Style/SingleLineDoEnd:
```
Comment (bbatsov) [inline config/default.yml:5189]: I think there's a style guide entry that you can refer here.

File: lib/rubocop/cop/style/single_line_do_end.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for single-line `do`...`end` blocks.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/single_line_do_end.rb:6]: You might also add a note about why the auto-correct doesn't replace `do/end` with `{}`, as perhaps this won't be clear for some people.

File: config/default.yml (hunk 1 above)
Comment (koic) [reply config/default.yml]: Sure! I've opened https://github.com/rubocop/ruby-style-guide/pull/922 to add a new rule for this one.

File: lib/rubocop/cop/style/single_line_do_end.rb (hunk 2 above)
Comment (koic) [reply lib/rubocop/cop/style/single_line_do_end.rb]: Indeed, I've added an explain for that.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**